monolith/views/restaurants.py

```diff
@@ -211,37 +222,95 @@ def my_reservations():
+        # update query
+        q = Restaurant.query.filter_by(id=session["RESTAURANT_ID"]).update(
+            {
+                "name": request.form.get("name"),
+                "lat": request.form.get("lat"),
+                "lon": request.form.get("lon"),
+                "covid_measures": request.form.get("covid_measures"),
+            }
+        )
+        # if no resturant match the update query (session problem probably)
+        if q == 0:
+            message = "Some Errors occurs"
         else:
-            return redirect("/create_restaurant")
+            db.session.commit()
+            message = "Restaurant data has been modified."
+
+    # get the resturant info and fill the form
+    # this part is both for POST and GET requests
+    q = Restaurant.query.filter_by(id=session["RESTAURANT_ID"]).first()
+    if q is not None:
+        print(q.covid_measures)
+        form = RestaurantForm(obj=q)
+        form2 = RestaurantTableForm()
+        tables = RestaurantTable.query.filter_by(restaurant_id=session["RESTAURANT_ID"])
+        return render_template(
+            "my_restaurant_data.html",
+            form=form,
+            only=["name", "lat", "lon", "covid_measures"],
+            tables=tables,
+            form2=form2,
+            message=message,
+        )
+    else:
+        return redirect("/create_restaurant")
+
 
 @restaurants.route("/mytables", methods=["GET", "POST"])
 @login_required
 @roles_allowed(roles=["OPERATOR"])
 def my_tables():
     if request.method == "POST":
-        # TODO: Add logic
+        # insert the table with data provided by the form
+        table = RestaurantTable()
+        table.restaurant_id = session["RESTAURANT_ID"]
+        table.max_seats = request.form.get("capacity")
+        table.name = request.form.get("name")
+        db.session.add(table)
+        db.session.commit()
+        ##
         return redirect("/my_restaurant_data")
+
     elif request.method == "GET":
-        # TODO: Delete logic, you have table id in GET ?id=
+        # delete the table specified by the get request
+        RestaurantTable.query.filter_by(id=request.args.get("id")).delete()
+        db.session.commit()
         return redirect("/my_restaurant_data")
 
 
+@restaurants.route("/my_restaurant_photogallery", methods=["GET", "POST"])
+@login_required
+@roles_allowed(roles=["OPERATOR"])
+def my_photogallery():
+    if request.method == "POST":
+        form = PhotoGalleryForm()
+        # add photo to the db
+        if form.validate_on_submit():
+            photo_gallery = PhotoGallery()
+            photo_gallery.caption = form.data["caption"]
+            photo_gallery.url = form.data["url"]
+            photo_gallery.restaurant_id = session["RESTAURANT_ID"]
+            db.session.add(photo_gallery)
+            db.session.commit()
+
+        return redirect("/my_restaurant_photogallery")
+    else:
+        photos = PhotoGallery.query.filter_by(
```

Comment:
If I'm on the page of the restaurant, I can click on the photo gallery and with a get request I will go inside this method but, for some reason, I don't see where you store the `RESTAURANT_ID`.

I receive this exception

```
Traceback (most recent call last):
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/vincent/.local/lib/python3.6/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/home/vincent/GithubUnipi/GoOutSafe_Primer2020/monolith/auth.py", line 35, in f
    return func(*args, **kwargs)
  File "/home/vincent/GithubUnipi/GoOutSafe_Primer2020/monolith/views/restaurants.py", line 308, in my_photogallery
    restaurant_id=session["RESTAURANT_ID"]
  File "/usr/local/lib/python3.6/dist-packages/werkzeug/local.py", line 377, in <lambda>
    __getitem__ = lambda x, i: x._get_current_object()[i]
  File "/home/vincent/.local/lib/python3.6/site-packages/flask/sessions.py", line 84, in __getitem__
    return super(SecureCookieSession, self).__getitem__(key)
KeyError: 'RESTAURANT_ID'
127.0.0.1 - - [27/Oct/2020 21:59:46] "GET /my_restaurant_photogallery?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1" 200 -
127.0.0.1 - - [27/Oct/2020 21:59:46] "GET /my_restaurant_photogallery?__debugger__=yes&cmd=resource&f=jquery.js HTTP/1.1" 200 -
127.0.0.1 - - [27/Oct/2020 21:59:46] "GET /my_restaurant_photogallery?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1" 200 -
127.0.0.1 - - [27/Oct/2020 21:59:46] "GET /my_restaurant_photogallery?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1" 200 -

```

Reply:
Fixed for now and tested it